src/content/app/genome-browser/hooks/useGenomeBrowserAnalytics.ts

```diff
@@ -0,0 +1,305 @@
+      category: 'track_settings',
+      label: selectedCog,
+      action:
+        'several_transcripts_' + (areSeveralTranscriptsShown ? 'on' : 'off')
+    });
+  };
+
+  const trackShowTranscriptsIdToggle = (
+    selectedCog: string,
+    shouldShowTranscriptIds: boolean
+  ) => {
+    sendTrackEvent({
+      category: 'track_settings',
+      label: selectedCog,
+      action: 'transcript_labels_' + (shouldShowTranscriptIds ? 'on' : 'off')
+    });
+  };
+
+  const trackApplyToAllInTackSettings = (
+    selectedCog: string,
+    shouldApplyToAll: boolean
+  ) => {
+    sendTrackEvent({
+      category: 'track_settings',
+      label: selectedCog,
+      action: 'apply_to_all - ' + (shouldApplyToAll ? 'unselected' : 'selected')
+    });
+  };
+
+  const trackTrackPanelTabChange = (selectedTrackPanelTab: TrackSet) => {
+    sendTrackEvent({
+      category: 'track_panel_tab',
+      label: selectedTrackPanelTab,
+      action: 'selected'
+    });
+  };
+
+  const trackTrackPanelSectionToggled = (sectionName: string) => {
+    sendTrackEvent({
+      category: 'track_panel',
+      label: sectionName,
+      action: 'section_toggled'
+    });
+  };
+
+  const trackFocusObjectReset = () => {
+    sendTrackEvent({
+      category: 'reset_icon',
+      action: 'clicked'
+    });
+  };
+
+  const trackSidebarModalViewToggle = (view: string) => {
+    sendTrackEvent({
+      category: 'sidebar_toolstrip',
+      action: 'modal_opened',
+      label: view
+    });
+  };
+
+  const trackDrawerOpened = (drawerView: string) => {
+    sendTrackEvent({
+      category: 'track_drawer',
+      action: 'opened',
+      label: drawerView
+    });
+  };
+
+  const trackDrawerSequenceViewed = (sequenceType: string) => {
+    sendTrackEvent({
+      category: 'track_drawer',
+      action: 'sequence_viewed',
+      label: sequenceType
+    });
+  };
+
+  const trackDrawerSequenceCopied = (sequenceType: string) => {
+    sendTrackEvent({
+      category: 'track_drawer',
+      action: 'sequence_copied',
+      label: sequenceType
+    });
+  };
+
+  const trackDrawerSequenceDownloaded = (selectedOptions: string) => {
+    sendTrackEvent({
+      category: 'track_drawer',
+      action: 'sequence_downloaded',
+      label: selectedOptions
+    });
+  };
+
+  const trackSpeciesChangeLinkClicked = () => {
+    sendTrackEvent({
+      category: 'app_bar',
+      action: 'change_link_clicked'
+    });
+  };
+
+  const trackGenomeChanged = () => {
+    sendTrackEvent({
+      category: 'app_bar',
+      action: 'change_genome'
+    });
+  };
+
+  const trackTrackPanelToggled = (isOpened: boolean) => {
+    sendTrackEvent({
+      category: 'track_panel',
+      action: isOpened ? 'opened' : 'closed'
+    });
+  };
+
+  const trackTrackVisibilityToggled = (trackName: string, status: boolean) => {
+    sendTrackEvent({
+      category: 'regular_track_visibility',
+      action: 'turned_' + status ? 'on' : 'off',
+      label: trackName
+    });
+  };
```

Comment:
- Why do we track `regular_track_visibility` and `focus_track_visibility` as different categories? Why isn't it a single category, with the focus track being called "focus"?
- When the focus gene track will ghost out its transcript (if that ever happens), it will always remain visible. Should be something to consider when designing our reporters.